instrumentation-api-semconv/src/testStableHttpSemconv/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractorStableSemconvTest.java

```diff
@@ -219,4 +222,82 @@ public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
           arguments("http", "42", "tcp", "tcp"));
     }
   }
+
+  @ParameterizedTest
+  @ValueSource(
+      strings = {"CONNECT", "DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT", "TRACE"})
+  void shouldExtractKnownMethods(String requestMethod) {
+    Map<String, String> request = new HashMap<>();
+    request.put("method", requestMethod);
+
+    AttributesExtractor<Map<String, String>, Map<String, String>> extractor =
+        HttpClientAttributesExtractor.create(
+            new TestHttpClientAttributesGetter(), new TestNetClientAttributesGetter());
+
+    AttributesBuilder attributes = Attributes.builder();
+    extractor.onStart(attributes, Context.root(), request);
+    extractor.onEnd(attributes, Context.root(), request, emptyMap(), null);
+
+    assertThat(attributes.build()).containsEntry(HttpAttributes.HTTP_REQUEST_METHOD, requestMethod);
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"get", "PURGE", "Get", "not a method really"})
```

Comment:
Nit: This is kinda mixing the "not-a-method" testing with "methods are case sensitive". I'd split the case sensitive stuff into its own test.